### ISSUE TITLE
Running MPS model crash '_mtlIOGPUCommandBufferStorageRebaseShmemHeader'

### ISSUE THREAD

**Issue by inspirit** [pytorch/pytorch#77843]
### 🐛 Describe the bug

Running model using 'mps' device resulted in some internal crash
`Python[4210:2560233] failed assertion false at line 567 in _mtlIOGPUCommandBufferStorageRebaseShmemHeader`
without any additional output/info.

### Versions

PyTorch version: 1.12.0.dev20220518
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.3.1 (arm64)
GCC version: Could not collect
Clang version: 13.1.6 (clang-1316.0.21.2.5)
CMake version: version 3.23.1
Libc version: N/A

Python version: 3.9.12 (main, Mar 26 2022, 15:44:31)  [Clang 13.1.6 (clang-1316.0.21.2)] (64-bit runtime)
Python platform: macOS-12.3.1-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.22.3
[pip3] torch==1.12.0.dev20220518
[pip3] torchaudio==0.11.0
[pip3] torchvision==0.12.0
[conda] Could not collect

**Comment by albanD** [comment]
Hi,

Would you have any small code sample that we can use to reproduce this please?

**Comment by inspirit** [comment]
hey that is a big project so its hard to tell what exactly cause internal issue, after some print and run trails I seem to be pin-point issue to following layer:
`nn.Conv1d(dim_in, dim_out, kernel_size=3, padding=1, padding_mode='circular', bias=False)`
what is strange it runs successfully few loop iterations and then crashes with the above error.
if i remove `padding_mode` argument it seems to run fine

**Comment by albanD** [comment]
Ok!
Can you share what are the values of `dim_in`/`dim_out` and the size of the Tensors you pass as input?

**Comment by inspirit** [comment]
Yup!
dim_in = 8
dim_out = 128
input tensors: (32, 8, 64) or (4, 8, 64)

**Comment by razarmehr** [comment]
Hi @inspirit . I tried the Conv1d with the input sizes you provided (test case below), and I'm unable to reproduce the issue. Please try with the latest nightly build and re-open the issue if you still see the crash. Thank you.

`def test_conv1d_circular_padding(self):
        y_cpu = torch.randn(32, 8, 64)
        conv_cpu = nn.Conv1d(8, 128, kernel_size=3, padding=1, padding_mode='circular', bias=False)
        conv_gpu = copy.deepcopy(conv_cpu).to(device='mps')
        x_cpu = conv_cpu(y_cpu)

        y_gpu = y_cpu.to(device='mps')
        x_gpu = conv_gpu(y_gpu)
        self.assertEqual(x_cpu, x_gpu.cpu())`
